torch/linalg/__init__.py

```diff
@@ -345,14 +361,28 @@
 """)
 
 eigvals = _add_docstr(_linalg.linalg_eigvals, r"""
-linalg.eigvals(input, *, out=None) -> Tensor
+linalg.eigvals(A, *, out=None) -> Tensor
+
+Computes the eigenvalues of a square matrix.
 
-Computes the eigenvalues of a square matrix. Supports batched inputs.
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+the eigenvalues of a square matrix :math:`A \in \mathbb{K}^{n \times n}` are defined
+as the roots (counted with multiplicity) of the polynomial of degree `n`
+
+.. math::
 
-Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
-The output tensor will always be complex-valued, even when :attr:`input` is real.
+    p(\lambda) = \operatorname{det}(A - \lambda \mathrm{I}_n)
 
-.. note:: The eigenvalues of a real matrix may be complex.
+where :math:`\mathrm{I}_n` is the `n`-dimensional identity matrix.
+
+Supports inputs of float, double, cfloat and cdouble dtypes.
+Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
+
+The returned tensor will always be complex-valued, even when :attr:`A` is real.
+
+.. note:: The eigenvalues of a real matrix may be complex, as the roots of a real polynomial may be complex.
```

Comment:
I would combine this note with the preceding sentence by extending it with:

"... even when :attr:\`A\` is real, because the roots of a real polynomial may be complex."

Reply:
The previous sentence has now been moved to the `Returns:`  section.